global_logger_test.go

```diff
@@ -193,3 +196,40 @@ func TestChainingSetup(t *testing.T) {
 		assertLogEntryContains(t, bytes.NewReader(b), "number", float64(42))
 	}
 }
+
+func TestGlobalLoggerFromEnv(t *testing.T) {
+	const (
+		logmsg  = "Some log"
+		envName = "LOG_LEVEL"
+	)
+
+	buf := &bytes.Buffer{}
+	oldOutput := globalLogger.output
+	oldNowFunc := globalLogger.now
+	defer func() {
+		SetOutput(oldOutput)
+		SetNowFunc(oldNowFunc)
+		SetReportCaller(true)
+		SetLevel(globalLogger.level)
+	}()
+	{
+		SetOutput(buf)
+		SetNowFunc(mockNowFunc)
+		SetReportCaller(false)
+	}
+
+	ConfigureGlobalLogger(WithLevelFromEnv(envName), WithOutput(buf))
+	Info(logmsg)
+	decodeLogToMap(t, buf)
+
+	assert.NotContains(t, buf.String(), logmsg)
+
+	// Test that it reads the var correctly to a Level in the Global Config
+	buf.Reset() // reset the buffer
+
+	os.Setenv("LOG_LEVEL", "info")
```

Comment:
```suggestion
	t.Setenv(envName, "info")
```

By using `t.Setenv` instead of `os.Setenv`, the env-var is automatically reset after the test is done. This will prevent us from having the env-var still set in another test that might expect that it is not set.